## Opening a French Restaurant in Chicago, IL, USA

This Jupyter Notebook run in Python 3.7 serves to collect, prepare and analyze data about all 77 communities in Chicago to help a French Restaurant owner decide which communities to prioritize when searching for a location to open his first restaurant unit in the city.

First, let's start by importing all libraries we'll need for this analysis:

In [124]:
import numpy as np # library to handle data in a vectorized manner
import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe
import json # library to handle JSON files

!pip install geopy
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values
!pip install folium
import folium # map rendering library
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler

print('Libraries imported.')

Libraries imported.


## 1. Data requirements and collection

In this data science project, we will run a K-means Clustering maching learning model taking into account the following 5 features about Chicago communities:
1. Population
2. Population Density
3. Number of restaurants
4. Ratio people/restaurant
5. Frequency of each restaurant category

Population and Population Density are features that we will collect from a Wikipedia page. Restaurant information will be collected using Foursquare Places API. Finally the ratio Population/Restaurant will be calculated from the data collected from both Wikipedia and Foursquare.

### 1a. Collecting list of communities, their population and population density.

In [125]:
#Gets and reads table from the Wikipedia webpage
url='https://en.wikipedia.org/wiki/Community_areas_in_Chicago'
source = requests.get(url)
source_data = pd.read_html(source.text)[0]
source_data

Number[8]                 Name[8]  2017[9] Area (sq mi.)[10]  Area (km2)  \
0         01             Rogers Park    55062              1.84        4.77   
1         02              West Ridge    76215              3.53        9.14   
2         03                  Uptown    57973              2.32        6.01   
3         04          Lincoln Square    41715              2.56        6.63   
4         05            North Center    35789              2.05        5.31   
5         06               Lake View   100470              3.12        8.08   
6         07            Lincoln Park    67710              3.16        8.18   
7         08         Near North Side    88893              2.74        7.10   
8         09             Edison Park    11605              1.13        2.93   
9         10            Norwood Park    37089              4.37       11.32   
10        11          Jefferson Park    26808              2.33        6.03   
11        12             Forest Glen    19019              3.20        8.29   
12        13              North Park    18842              2.52        6.53   
13        14             Albany Park    51992              1.92        4.97   
14        15            Portage Park    64307              3.95       10.23   
15        16             Irving Park    54606              3.21        8.31   
16        17                 Dunning    43689              3.72        9.63   
17        18               Montclare    13830              0.99        2.56   
18        19          Belmont Cragin    79910              3.91       10.13   
19        20                 Hermosa    24144              1.17        3.03   
20        21                Avondale    37368              1.98        5.13   
21        22            Logan Square    73046              3.59        9.30   
22        23           Humboldt Park    56427              3.60        9.32   
23        24               West Town    84502              4.58       11.86   
24        25                  Austin    95260              7.15       18.52   
25        26      West Garfield Park    17163              1.28        3.32   
26        27      East Garfield Park    19996              1.93        5.00   
27        28          Near West Side    62872              5.69       14.74   
28        29          North Lawndale    35947              3.21        8.31   
29        30          South Lawndale    74851              4.59       11.89   
30        31         Lower West Side    32888              2.93        7.59   
31        32          (The) Loop[11]    35880              1.65        4.27   
32        33         Near South Side    23620              1.78        4.61   
33        34           Armour Square    13455              1.00        2.59   
34        35                 Douglas    20781              1.65        4.27   
35        36                 Oakland     6645              0.58        1.50   
36        37             Fuller Park     2439              0.71        1.84   
37        38         Grand Boulevard    22313              1.74        4.51   
38        39                 Kenwood    17189              1.04        2.69   
39        40         Washington Park    11502              1.52        3.94   
40        41               Hyde Park    26827              1.61        4.17   
41        42                Woodlawn    23268              2.07        5.36   
42        43             South Shore    50418              2.93        7.59   
43        44                 Chatham    31120              2.95        7.64   
44        45             Avalon Park     9985              1.25        3.24   
45        46           South Chicago    28263              3.34        8.65   
46        47                Burnside     2254              0.61        1.58   
47        48         Calumet Heights    13188              1.75        4.53   
48        49                Roseland    42433              4.82       12.48   
49        50                 Pullman     6613        

In [126]:
# checking if there are any duplicate communities.
#There are 78 rows in the table, so if the code below yields value 78, there are no duplicates.
source_data['Name[8]'].nunique()

78

By observing the table scrapped above, we want to make a few corrections:
1. Drop all columns except for Name[8], which represents Community Name, 2017[9] which represents Population data in 2017, and 2017 Density in km2.
2. Correct the name of the Loop to remove the paranthesis and the [11] footnote mark
3. Delete the last row with totals

In [127]:
source_data=source_data.drop(['Number[8]','Area (sq mi.)[10]','Area (km2)','2017density (/sq mi.)'], axis=1)
source_data=source_data.drop([77])
source_data['Name[8]']=source_data.replace(["(The) Loop[11]"],"Loop")
source_data

Name[8]  2017[9]  2017density (/km2)
0              Rogers Park    55062            11554.11
1               West Ridge    76215             8336.20
2                   Uptown    57973             9648.06
3           Lincoln Square    41715             6291.50
4             North Center    35789             6740.59
5                Lake View   100470            12433.23
6             Lincoln Park    67710             8273.10
7          Near North Side    88893            12526.20
8              Edison Park    11605             1635.30
9             Norwood Park    37089             3276.92
10          Jefferson Park    26808             4442.33
11             Forest Glen    19019             2294.78
12              North Park    18842             2886.88
13             Albany Park    51992            10455.33
14            Portage Park    64307             6285.84
15             Irving Park    54606             6568.06
16                 Dunning    43689             4534.52
17               Montclare    13830             5393.73
18          Belmont Cragin    79910             7890.90
19                 Hermosa    24144             7967.57
20                Avondale    37368             7286.80
21            Logan Square    73046             7856.05
22           Humboldt Park    56427             6051.83
23               West Town    84502             7123.67
24                  Austin    95260             5144.07
25      West Garfield Park    17163             5177.09
26      East Garfield Park    19996             4000.26
27          Near West Side    62872             4266.26
28          North Lawndale    35947             4323.74
29          South Lawndale    74851             6296.33
30         Lower West Side    32888             4333.83
31                    Loop    35880             8395.97
32         Near South Side    23620             5123.44
33           Armour Square    13455             5195.00
34                 Douglas    20781             4862.78
35                 Oakland     6645             4423.53
36             Fuller Park     2439             1326.34
37         Grand Boulevard    22313             4951.20
38                 Kenwood    17189             6381.45
39         Washington Park    11502             2921.68
40               Hyde Park    26827             6433.52
41                Woodlawn    23268             4340.01
42             South Shore    50418             6643.86
43                 Chatham    31120             4073.05
44             Avalon Park     9985             3084.18
45           South Chicago    28263             3267.19
46                Burnside     2254             1426.68
47         Calumet Heights    13188             2909.67
48                Roseland    42433             3399.06
49                 Pullman     6613             1204.38
50           South Deering    14614              517.66
51               East Side    23737             3075.47
52            West Pullman    27742             3008.78
53               Riverdale     7394              808.74
54               Hegewisch     9418              693.95
55          Garfield Ridge    36396             3322.12
56          Archer Heights    13142             2524.46
57           Brighton Park    44813             6361.18
58           McKinley Park    15767             4317.50
59              Bridgeport    33637             6214.03
60                New City    39561             3162.44
61             West Elsdon    19237             6348.25
62               Gage Park    40873             7173.25
63                Clearing    25891             3920.22
64               West Lawn    33108             4333.24
65            Chicago Lawn    53098             5807.72
66          West Englewood    29929             3668.46
67               Englewood    25075             3153.59
68  Greater Grand Crossing    31766             3454.91
69                 Ashburn    43792             3479.05
70          Auburn Gresham    46278        

Now let's rename the columns to names that makes sense:

In [128]:
source_data=source_data.rename(columns={"Name[8]": "Community", "2017[9]": "Population", "2017density (/km2)": "Density (/km2)"})
source_data.head()

Community  Population  Density (/km2)
0     Rogers Park       55062        11554.11
1      West Ridge       76215         8336.20
2          Uptown       57973         9648.06
3  Lincoln Square       41715         6291.50
4    North Center       35789         6740.59

In [129]:
# chacking data types
source_data.dtypes

Community          object
Population          int64
Density (/km2)    float64
dtype: object

### 1b. Collecting Restaurant information accross communities

We want to use the Foursquare API to get the names and categories of all restaurants in all Chicago communities. To do so, we first need to obtain latitude and longitude coordinates for all 77 communities in Chicago.

In [130]:
geolocator = Nominatim(user_agent="coord_explorer")

# creates a dataframe to collect coordinates for all communities
columns_names = ['Community','Latitude','Longitude']
community_data = pd.DataFrame(columns = columns_names)

for comm in source_data['Community']:
    community = comm
    geolocator = Nominatim(user_agent="coord_explorer")
    address = comm + ", Chicago, IL"
    location = geolocator.geocode(address)
    latitude = location.latitude
    longitude = location.longitude
    
    community_data = community_data.append({'Community': community,
                                            'Latitude': latitude,
                                            'Longitude': longitude,}, ignore_index=True)

community_data

Community   Latitude  Longitude
0              Rogers Park  42.010531 -87.670748
1               West Ridge  42.003548 -87.696243
2                   Uptown  41.966630 -87.655546
3           Lincoln Square  41.975990 -87.689616
4             North Center  41.956107 -87.679160
5                Lake View  41.947050 -87.655423
6             Lincoln Park  41.939945 -87.636120
7          Near North Side  41.900033 -87.634497
8              Edison Park  42.006113 -87.813992
9             Norwood Park  41.985590 -87.800582
10          Jefferson Park  41.969738 -87.763118
11             Forest Glen  41.991752 -87.751674
12              North Park  41.980587 -87.720892
13             Albany Park  41.971937 -87.716174
14            Portage Park  41.957809 -87.765059
15             Irving Park  41.953365 -87.736447
16                 Dunning  41.952809 -87.796449
17               Montclare  41.925309 -87.800893
18          Belmont Cragin  41.931698 -87.768670
19                 Hermosa  41.928643 -87.734502
20                Avondale  41.938921 -87.711168
21            Logan Square  41.928568 -87.706793
22           Humboldt Park  41.900828 -87.723959
23               West Town  41.901421 -87.686166
24                  Austin  41.887876 -87.764851
25      West Garfield Park  41.880588 -87.729223
26      East Garfield Park  41.880866 -87.702833
27          Near West Side  41.880066 -87.666762
28          North Lawndale  41.860032 -87.718388
29          South Lawndale  41.843644 -87.712554
30         Lower West Side  41.854200 -87.665609
31                    Loop  41.881609 -87.629457
32         Near South Side  41.856700 -87.624774
33           Armour Square  41.840231 -87.632986
34                 Douglas  41.834857 -87.617954
35                 Oakland  41.824051 -87.605117
36             Fuller Park  41.818089 -87.632551
37         Grand Boulevard  41.813923 -87.617272
38                 Kenwood  41.809144 -87.597991
39         Washington Park  41.792534 -87.618105
40               Hyde Park  41.794446 -87.593924
41                Woodlawn  41.779479 -87.599493
42             South Shore  41.758728 -87.575283
43                 Chatham  41.741145 -87.612548
44             Avalon Park  41.745035 -87.588658
45           South Chicago  41.734587 -87.551322
46                Burnside  41.730035 -87.596714
47         Calumet Heights  41.730035 -87.579213
48                Roseland  41.701979 -87.625881
49                 Pullman  41.692218 -87.606604
50           South Deering  41.692236 -87.571703
51               East Side  41.713569 -87.532781
52            West Pullman  41.671775 -87.638358
53               Riverdale  41.667835 -87.604964
54               Hegewisch  41.653646 -87.546988
55          Garfield Ridge  41.803617 -87.745489
56          Archer Heights  41.811422 -87.726165
57           Brighton Park  41.818922 -87.698942
58           McKinley Park  41.831700 -87.673664
59              Bridgeport  41.837938 -87.651028
60                New City  41.807533 -87.656440
61             West Elsdon  41.793922 -87.724498
62               Gage Park  41.795033 -87.696164
63                Clearing  41.780588 -87.773388
64               West Lawn  41.772811 -87.722275
65            Chicago Lawn  41.775033 -87.696441
66          West Englewood  41.778089 -87.666718
67               Englewood  41.779756 -87.645884
68  Greater Grand Crossing  41.766886 -87.620845
69                 Ashburn  41.747533 -87.711163
70          Auburn Gresham  41.743387 -87.656042
71                 Beverly  41.718153 -87.671767
72      Washington Heights  41.706423 -87.656160
73         Mount Greenwood  41.698089 -87.708662
74             Morgan Park  41.690312 -87.666716
75                  O'Hare  41.973101 -87.906768
76               Edgewater  41.983369 -87.663952

In [131]:
# merge with the remaining information of source_data:
community_data = community_data.merge(source_data, on='Community', how = 'left')
community_data

Community   Latitude  Longitude  Population  Density (/km2)
0              Rogers Park  42.010531 -87.670748       55062        11554.11
1               West Ridge  42.003548 -87.696243       76215         8336.20
2                   Uptown  41.966630 -87.655546       57973         9648.06
3           Lincoln Square  41.975990 -87.689616       41715         6291.50
4             North Center  41.956107 -87.679160       35789         6740.59
5                Lake View  41.947050 -87.655423      100470        12433.23
6             Lincoln Park  41.939945 -87.636120       67710         8273.10
7          Near North Side  41.900033 -87.634497       88893        12526.20
8              Edison Park  42.006113 -87.813992       11605         1635.30
9             Norwood Park  41.985590 -87.800582       37089         3276.92
10          Jefferson Park  41.969738 -87.763118       26808         4442.33
11             Forest Glen  41.991752 -87.751674       19019         2294.78
12              North Park  41.980587 -87.720892       18842         2886.88
13             Albany Park  41.971937 -87.716174       51992        10455.33
14            Portage Park  41.957809 -87.765059       64307         6285.84
15             Irving Park  41.953365 -87.736447       54606         6568.06
16                 Dunning  41.952809 -87.796449       43689         4534.52
17               Montclare  41.925309 -87.800893       13830         5393.73
18          Belmont Cragin  41.931698 -87.768670       79910         7890.90
19                 Hermosa  41.928643 -87.734502       24144         7967.57
20                Avondale  41.938921 -87.711168       37368         7286.80
21            Logan Square  41.928568 -87.706793       73046         7856.05
22           Humboldt Park  41.900828 -87.723959       56427         6051.83
23               West Town  41.901421 -87.686166       84502         7123.67
24                  Austin  41.887876 -87.764851       95260         5144.07
25      West Garfield Park  41.880588 -87.729223       17163         5177.09
26      East Garfield Park  41.880866 -87.702833       19996         4000.26
27          Near West Side  41.880066 -87.666762       62872         4266.26
28          North Lawndale  41.860032 -87.718388       35947         4323.74
29          South Lawndale  41.843644 -87.712554       74851         6296.33
30         Lower West Side  41.854200 -87.665609       32888         4333.83
31                    Loop  41.881609 -87.629457       35880         8395.97
32         Near South Side  41.856700 -87.624774       23620         5123.44
33           Armour Square  41.840231 -87.632986       13455         5195.00
34                 Douglas  41.834857 -87.617954       20781         4862.78
35                 Oakland  41.824051 -87.605117        6645         4423.53
36             Fuller Park  41.818089 -87.632551        2439         1326.34
37         Grand Boulevard  41.813923 -87.617272       22313         4951.20
38                 Kenwood  41.809144 -87.597991       17189         6381.45
39         Washington Park  41.792534 -87.618105       11502         2921.68
40               Hyde Park  41.794446 -87.593924       26827         6433.52
41                Woodlawn  41.779479 -87.599493       23268         4340.01
42             South Shore  41.758728 -87.575283       50418         6643.86
43                 Chatham  41.741145 -87.612548       31120         4073.05
44             Avalon Park  41.745035 -87.588658        9985         3084.18
45           South Chicago  41.734587 -87.551322       28263         3267.19
46                Burnside  41.730035 -87.596714        2254         1426.68
47         Calumet Heights  41.730035 -87.579213       13188         2909.67
48                Roseland  41.701979 -87.625881       42433         3399.06
49                 Pullman  41.692218 -87.606604        6613         1204.38
50           South Deering  41.692236 -87.571703       14614          517.66
51           

Let's visualize all communities in a Chicago map

In [132]:
# Get coordinates (longitude and latitude) of Chicago

address = 'Chicago, IL'

geolocator = Nominatim(user_agent="geo_explorer")
location = geolocator.geocode(address)
chicago_latitude = location.latitude
chicago_longitude = location.longitude
print('The geograpical coordinates of Chicago, IL are lat:{}, lng:{}.'.format(chicago_latitude, chicago_longitude))

The geograpical coordinates of Chicago, IL are lat:41.8755616, lng:-87.6244212.


In [133]:
# create map of Chicago
chicago_map = folium.Map(location=[chicago_latitude, chicago_longitude], zoom_start=10)

# add neighbordhood markers to map
for lat, lng, label in zip(community_data['Latitude'], community_data['Longitude'], community_data['Community']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(chicago_map)  
    
chicago_map

Now let's get Restaurant data in each community from Foursquare. We will start with a test using the first listed community in our community_data dataset

In [134]:
CLIENT_ID = 'S533B0JYPPDULC0CWZPUGPXOI3PBMB1MQZVRLW4XQNJPXEVM'
CLIENT_SECRET = 'MUGAY5GYI2CD45CNFRYUTSNZHBN2UO2J20DEZGQNACI5JDP3'
ACCESS_TOKEN = 'MMFSJNNH0Z05PDMR5CYAC40OSTDHIQOEKU5U2SN52ZFJBKFR'
VERSION = '20180604'
LIMIT = 100

In [135]:
# Set parameters to only get results for the first community in our community_data DataFrame
latitude = community_data['Latitude'][0]
longitude = community_data['Longitude'][0]
search_query = 'Restaurant'
radius = 500

url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&oauth_token={}&v={}&ll={},{}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, ACCESS_TOKEN, VERSION, latitude, longitude, search_query, radius, LIMIT)
results = requests.get(url).json()['response']['venues']
results

[{'id': '4b75fc9bf964a52070342ee3',
  'name': 'Hong Kong Restaurant',
  'location': {'address': '6958 N Clark St',
   'lat': 42.00814437866211,
   'lng': -87.67413330078125,
   'labeledLatLngs': [{'label': 'entrance',
     'lat': 42.008298,
     'lng': -87.673924},
    {'label': 'display', 'lat': 42.00814437866211, 'lng': -87.67413330078125}],
   'distance': 386,
   'postalCode': '60626',
   'cc': 'US',
   'city': 'Chicago',
   'state': 'IL',
   'country': 'United States',
   'formattedAddress': ['6958 N Clark St', 'Chicago, IL 60626']},
  'categories': [{'id': '4bf58dd8d48988d145941735',
    'name': 'Chinese Restaurant',
    'pluralName': 'Chinese Restaurants',
    'shortName': 'Chinese',
    'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/asian_',
     'suffix': '.png'},
    'primary': True}],
  'referralId': 'v-1609210223',
  'hasPerk': False},
 {'id': '4df0fa23d4c04d0392c87e45',
  'name': 'Taqueria & Restaurant Cd. Hidalgo',
  'location': {'address': '7104 N Clark S

In [136]:
# Get the 3 data fields we want: venue id, venue name, and venue category

print("Venue ID: ", results[0]['id'])
print("Venue Name: ", results[0]['name'])
print("Category: ", results[0]['categories'][0]['name'])

Venue ID:  4b75fc9bf964a52070342ee3
Venue Name:  Hong Kong Restaurant
Category:  Chinese Restaurant


Now based on the successful test above, let's write a function that gets Restaurant information from Foursquare for all 77 communities in Chicago

In [137]:
search_query = 'Restaurant'
radius = 1000

restaurants_list=[]
for comm, lat, lng in zip(community_data['Community'], community_data['Latitude'], community_data['Longitude']):
                    
    # create the API request URL
    url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&oauth_token={}&v={}&ll={},{}&query={}&radius={}&limit={}'.format(
    CLIENT_ID, CLIENT_SECRET, ACCESS_TOKEN, VERSION, lat, lng, search_query, radius, LIMIT)
        
    # try to make the GET request
    try:
        results = requests.get(url).json()['response']['venues']
        restaurants_list.append([(comm, v['id'], v['name'], v['categories'][0]['name']) for v in results])
        restaurants_data = pd.DataFrame([item for venue_list in restaurants_list for item in venue_list])
        restaurants_data.columns = ['Community','Venue ID','Venue Name','Category'] 
                              
    except:
        print("Failed to retrieve data for: ", comm) # alert in case Foursquare doesn't return any information for any community
        
restaurants_data

Failed to retrieve data for:  West Ridge
Failed to retrieve data for:  Lincoln Square
Failed to retrieve data for:  Lake View
Failed to retrieve data for:  Near North Side
Failed to retrieve data for:  Albany Park
Failed to retrieve data for:  Logan Square
Failed to retrieve data for:  Archer Heights
Failed to retrieve data for:  New City
Failed to retrieve data for:  Chicago Lawn
Failed to retrieve data for:  Edgewater


Community                  Venue ID  \
0               Rogers Park  4b75fc9bf964a52070342ee3   
1               Rogers Park  4df0fa23d4c04d0392c87e45   
2               Rogers Park  5a03af34b5461848270c64b6   
3               Rogers Park  4f44ff7c19836ed00198048d   
4               Rogers Park  4e4e317abd4101d0d7a46b18   
5               Rogers Park  4f32b71219836c91c7f29557   
6               Rogers Park  50900127e4b061e9d64fb0f9   
7               Rogers Park  4b721efbf964a520b36f2de3   
8               Rogers Park  4f43b68719834bc91f586d42   
9               Rogers Park  4e4e5cbdbd4101d0d7a881b4   
10              Rogers Park  4f32898419836c91c7e17db5   
11              Rogers Park  4f325cdf19836c91c7d0605f   
12              Rogers Park  4d7fff51208a548106f26c67   
13              Rogers Park  4b904e30f964a520f18233e3   
14              Rogers Park  4a4e5340f964a5204cae1fe3   
15              Rogers Park  4c4ce75ff96ac9b6d68fce98   
16              Rogers Park  4ac8dfc0f964a520fcbc20e3   
17              Rogers Park  4a902d3af964a5205a1620e3   
18              Rogers Park  4b55e587f964a52021f527e3   
19              Rogers Park  4b5e461af964a5202c8729e3   
20                   Uptown  4e1cef051f6e47e5e7936742   
21                   Uptown  4bc22170b492d13ac204a760   
22                   Uptown  4ad12520f964a5200ddd20e3   
23                   Uptown  4dbca6015da3ff58ec51c9d5   
24                   Uptown  4ad7d628f964a520390f21e3   
25                   Uptown  4e4e237ebd4101d0d7a3217a   
26                   Uptown  4ab16f42f964a520786920e3   
27                   Uptown  4f3261ce19836c91c7d24d2c   
28                   Uptown  4a589eddf964a52098b71fe3   
29                   Uptown  4df2c96ec65bf55ee529bb99   
30                   Uptown  4bdb88e63904a59319064b9e   
31                   Uptown  4f32698719836c91c7d54aa9   
32                   Uptown  4b3fbb6cf964a52078ad25e3   
33                   Uptown  5bb57200f4b525003979d084   
34                   Uptown  4fec3404e4b0c8b85f977b07   
35                   Uptown  4beb293a62c0c9280bcae1d4   
36                   Uptown  4cbf383b020d4688432c6138   
37                   Uptown  4bc385234cdfc9b669789721   
38                   Uptown  4e4e3110bd4101d0d7a46189   
39                   Uptown  4f99c36fe4b0e88398a2805e   
40                   Uptown  531bb962498ec8d7fb5c6628   
41                   Uptown  4f32614019836c91c7d21661   
42                   Uptown  5a410ba506fb603df129dcba   
43                   Uptown  4e4e19d8bd4101d0d7a244c6   
44                   Uptown  5bfb64cee97dfb002cc1f2d0   
45                   Uptown  4f43ac6119834bc91f582bad   
46                   Uptown  4f43fb9a19836ed001918f1c   
47                   Uptown  4f3210ed19833175d60d246a   
48                   Uptown  4f32043219833175d60a02b6   
49                   Uptown  4e1a1fffc65b6bfb591f3331   
50                   Uptown  4aee47bcf964a52057d321e3   
51                   Uptown  58817207ec980f27ba18800d   
52                   Uptown  4b076a5cf964a52054fd22e3   
53                   Uptown  4b2598a9f964a5204f7424e3   
54                   Uptown  4b11a5d9f964a5205a8123e3   
55                   Uptown  4a5cdeb0f964a520c0bc1fe3   
56                   Uptown  4a9dbaf5f964a520233920e3   
57                   Uptown  4aa3fa44f964a520a84420e3   
58                   Uptown  4aff226cf964a520b93422e3   
59                   Uptown  4c4f46e01d76c92816889591   
60                   Uptown  4a81ee9ff964a52003f81fe3   
61                   Uptown  4ec561a60e617a27fa1db65e   
62                   Uptown  49f0c67ff964a52056691fe3   
63                   Uptown  4ee6c6d8722e088261e5005e   
64                   Uptown  4a5a72f9f964a52057ba1fe3   
65                   Uptown  4bc9f18868f976b0fc315e83   
66                   Uptown  4b9c437ef964a520145a36e3   
67             North Center  4c2011d8b4e62d7fb0a7e093   
68             North Center  4a5a343df964a520e4b91fe3   
69             North Center 

In [138]:
#Checking the number of French Restaurants listed:
restaurants_data[restaurants_data['Category']=="French Restaurant"]

Community                  Venue ID                        Venue Name  \
90  Lincoln Park  5fd17b6c8fe7ac1127f63d97  Ann Sather Restaurant & Catering   

             Category  
90  French Restaurant

Interesting enough, we could only find one French restaurants listed accross all Chicago communities in Foursquare.

We also observe that some of the categories retrieved are unrelated to restaurants or eating out activities (for instance "Building" or "Office"). Let's remove from the dataset all datapoints that do not fall under dining experiences.

In [139]:
# retrieve a list of categories that do not contain the word Restaurant
non_rest = restaurants_data[restaurants_data['Category'].str.contains("Restaurant") == False]['Category'].unique()
non_rest

array(['Food', 'Wings Joint', 'Diner', 'Bakery', 'Café', 'Noodle House',
       'Breakfast Spot', 'Pizza Place', 'Pub', 'Bagel Shop', 'Steakhouse',
       'Park', 'Burger Joint', 'Miscellaneous Shop', 'BBQ Joint', 'Bar',
       'Food Court', 'Sandwich Place', 'Taco Place', 'Arcade',
       'Deli / Bodega', 'Lounge', 'Coffee Shop', 'Concert Hall',
       'Building', 'Grocery Store',
       'Residential Building (Apartment / Condo)', 'Office',
       'Hot Dog Joint', 'Conference Room', 'Wine Bar', 'Snack Place',
       'Event Space', 'Antique Shop', 'Market', 'Pop-Up Shop',
       'Gastropub', 'Gay Bar', 'Sports Bar'], dtype=object)

In [140]:
# prepares a list of categories to be removed
exclude_list = (['Park','Miscellaneous Shop','Arcarde','Lounge','Building','Concert Hall','Grocery Store',
                'Residential Building (Apartment / Condo)','Office','Conference Room','Event Space','Antique Shop','Market','Pop-Up Shop'])

# creates a new dataframe only taking data where Category does not fall in the exclude_list above
select_restaurants_data = restaurants_data[~restaurants_data['Category'].isin(exclude_list)]
select_restaurants_data

Community                  Venue ID  \
0               Rogers Park  4b75fc9bf964a52070342ee3   
1               Rogers Park  4df0fa23d4c04d0392c87e45   
2               Rogers Park  5a03af34b5461848270c64b6   
3               Rogers Park  4f44ff7c19836ed00198048d   
4               Rogers Park  4e4e317abd4101d0d7a46b18   
5               Rogers Park  4f32b71219836c91c7f29557   
6               Rogers Park  50900127e4b061e9d64fb0f9   
7               Rogers Park  4b721efbf964a520b36f2de3   
8               Rogers Park  4f43b68719834bc91f586d42   
9               Rogers Park  4e4e5cbdbd4101d0d7a881b4   
10              Rogers Park  4f32898419836c91c7e17db5   
11              Rogers Park  4f325cdf19836c91c7d0605f   
12              Rogers Park  4d7fff51208a548106f26c67   
13              Rogers Park  4b904e30f964a520f18233e3   
14              Rogers Park  4a4e5340f964a5204cae1fe3   
15              Rogers Park  4c4ce75ff96ac9b6d68fce98   
16              Rogers Park  4ac8dfc0f964a520fcbc20e3   
17              Rogers Park  4a902d3af964a5205a1620e3   
18              Rogers Park  4b55e587f964a52021f527e3   
19              Rogers Park  4b5e461af964a5202c8729e3   
20                   Uptown  4e1cef051f6e47e5e7936742   
21                   Uptown  4bc22170b492d13ac204a760   
22                   Uptown  4ad12520f964a5200ddd20e3   
23                   Uptown  4dbca6015da3ff58ec51c9d5   
24                   Uptown  4ad7d628f964a520390f21e3   
25                   Uptown  4e4e237ebd4101d0d7a3217a   
26                   Uptown  4ab16f42f964a520786920e3   
27                   Uptown  4f3261ce19836c91c7d24d2c   
28                   Uptown  4a589eddf964a52098b71fe3   
29                   Uptown  4df2c96ec65bf55ee529bb99   
30                   Uptown  4bdb88e63904a59319064b9e   
31                   Uptown  4f32698719836c91c7d54aa9   
32                   Uptown  4b3fbb6cf964a52078ad25e3   
33                   Uptown  5bb57200f4b525003979d084   
34                   Uptown  4fec3404e4b0c8b85f977b07   
35                   Uptown  4beb293a62c0c9280bcae1d4   
36                   Uptown  4cbf383b020d4688432c6138   
37                   Uptown  4bc385234cdfc9b669789721   
38                   Uptown  4e4e3110bd4101d0d7a46189   
39                   Uptown  4f99c36fe4b0e88398a2805e   
40                   Uptown  531bb962498ec8d7fb5c6628   
41                   Uptown  4f32614019836c91c7d21661   
42                   Uptown  5a410ba506fb603df129dcba   
43                   Uptown  4e4e19d8bd4101d0d7a244c6   
44                   Uptown  5bfb64cee97dfb002cc1f2d0   
45                   Uptown  4f43ac6119834bc91f582bad   
46                   Uptown  4f43fb9a19836ed001918f1c   
47                   Uptown  4f3210ed19833175d60d246a   
48                   Uptown  4f32043219833175d60a02b6   
49                   Uptown  4e1a1fffc65b6bfb591f3331   
50                   Uptown  4aee47bcf964a52057d321e3   
51                   Uptown  58817207ec980f27ba18800d   
52                   Uptown  4b076a5cf964a52054fd22e3   
53                   Uptown  4b2598a9f964a5204f7424e3   
54                   Uptown  4b11a5d9f964a5205a8123e3   
55                   Uptown  4a5cdeb0f964a520c0bc1fe3   
56                   Uptown  4a9dbaf5f964a520233920e3   
57                   Uptown  4aa3fa44f964a520a84420e3   
58                   Uptown  4aff226cf964a520b93422e3   
59                   Uptown  4c4f46e01d76c92816889591   
60                   Uptown  4a81ee9ff964a52003f81fe3   
61                   Uptown  4ec561a60e617a27fa1db65e   
62                   Uptown  49f0c67ff964a52056691fe3   
63                   Uptown  4ee6c6d8722e088261e5005e   
64                   Uptown  4a5a72f9f964a52057ba1fe3   
65                   Uptown  4bc9f18868f976b0fc315e83   
66                   Uptown  4b9c437ef964a520145a36e3   
67             North Center  4c2011d8b4e62d7fb0a7e093   
68             North Center  4a5a343df964a520e4b91fe3   
69             North Center 

We can now calculate the number of restaurants and similar venues in each community, which will be used in our Clustering model:

In [141]:
count_rest=select_restaurants_data.groupby('Community').count()
count_rest

Venue ID  Venue Name  Category
Community                                             
Armour Square                  6           6         6
Ashburn                        6           6         6
Auburn Gresham                 8           8         8
Austin                         6           6         6
Avalon Park                    5           5         5
Avondale                      31          31        31
Belmont Cragin                20          20        20
Beverly                        1           1         1
Bridgeport                    10          10        10
Brighton Park                  7           7         7
Burnside                       1           1         1
Calumet Heights                3           3         3
Chatham                        3           3         3
Clearing                       4           4         4
Douglas                        3           3         3
Dunning                        8           8         8
East Garfield Park             4           4         4
East Side                      3           3         3
Edison Park                    1           1         1
Englewood                      3           3         3
Forest Glen                    4           4         4
Fuller Park                    4           4         4
Gage Park                     10          10        10
Garfield Ridge                 6           6         6
Grand Boulevard                8           8         8
Greater Grand Crossing         4           4         4
Hegewisch                      1           1         1
Hermosa                       17          17        17
Humboldt Park                  6           6         6
Hyde Park                     14          14        14
Irving Park                   19          19        19
Jefferson Park                 7           7         7
Kenwood                        3           3         3
Lincoln Park                  15          15        15
Loop                          46          46        46
Lower West Side               21          21        21
McKinley Park                  7           7         7
Montclare                      5           5         5
Morgan Park                    1           1         1
Mount Greenwood                4           4         4
Near South Side               48          48        48
Near West Side                 9           9         9
North Center                  19          19        19
North Lawndale                 2           2         2
North Park                    13          13        13
Norwood Park                   7           7         7
O'Hare                         6           6         6
Oakland                        4           4         4
Portage Park                  10          10        10
Pullman                        4           4         4
Rogers Park                   20          20        20
Roseland                       3           3         3
South Chicago                  7           7         7
South Lawndale                16          16        16
South Shore                   11          11        11
Uptown                        47          47        47
Washington Park                3           3         3
West Elsdon                    9           9         9
West Englewood                 4           4         4
West Garfield Park             5           5         5
West Lawn                     14          14        14
West Pullman                   1           1         1
West Town                     25          25        25
Woodlawn                       5           5         5

In [142]:
count_rest.shape

(64, 3)

The Foursquare loop function could retrieve restaurant information for 64 of the 77 communities. Therefore, 13 communites were not identified in the search: West Ridge, Lincoln Square, Lake View, Near North Side, Albany Park, Logan Square, Archer Heights, New City, Chicago Lawn, Edgewater

## 2. Clustering

### 2a. Preparing data for clustering model

Before we run our Cluster modelling process, we need to normalize our data.

We will start by transforming our Categorical variable - Restaurant Categories - into a numerical frequency of categories in each community. All values will, by definition, fit a scale of float values ranging from 0 to 1.

In [143]:
# using the get_dummies function
categ_data = pd.get_dummies(select_restaurants_data['Category'])
categ_data['Community'] = select_restaurants_data['Community']
fixed_columns = [categ_data.columns[-1]] + list(categ_data.columns[:-1])
categ_data = categ_data[fixed_columns]
categ_data.head()

Community  African Restaurant  American Restaurant  Arcade  \
0  Rogers Park                   0                    0       0   
1  Rogers Park                   0                    0       0   
2  Rogers Park                   0                    0       0   
3  Rogers Park                   0                    0       0   
4  Rogers Park                   0                    0       0   

   Argentinian Restaurant  Asian Restaurant  BBQ Joint  Bagel Shop  Bakery  \
0                       0                 0          0           0       0   
1                       0                 0          0           0       0   
2                       0                 0          0           0       0   
3                       0                 0          0           0       0   
4                       0                 0          0           0       0   

   Bar  Breakfast Spot  Burger Joint  Café  Cajun / Creole Restaurant  \
0    0               0             0     0                          0   
1    0               0             0     0                          0   
2    0               0             0     0                          0   
3    0               0             0     0                          0   
4    0               0             0     0                          0   

   Cantonese Restaurant  Caribbean Restaurant  Chinese Restaurant  \
0                     0                     0                   1   
1                     0                     0                   0   
2                     0                     0                   0   
3                     0                     0                   0   
4                     0                     0                   0   

   Coffee Shop  Comfort Food Restaurant  Cuban Restaurant  Deli / Bodega  \
0            0                        0                 0              0   
1            0                        0                 0              0   
2            0                        0                 0              0   
3            0                        0                 0              0   
4            0                        0                 0              0   

   Dim Sum Restaurant  Diner  Eastern European Restaurant  \
0                   0      0                            0   
1                   0      0                            0   
2                   0      0                            0   
3                   0      0                            0   
4                   0      0                            0   

   Ethiopian Restaurant  Fast Food Restaurant  Food  Food Court  \
0                     0                     0     0           0   
1                     0                     0     0           0   
2                     0                     0     0           0   
3                     0                     0     1           0   
4                     0                     0     0           0   

   French Restaurant  Gastropub  Gay Bar  German Restaurant  Greek Restaurant  \
0                  0          0        0                  0                 0   
1                  0          0        0                  0                 0   
2                  0          0        0                  0                 0   
3                  0          0        0                  0                 0   
4                  0          0        0                  0                 0   

   Hot Dog Joint  Hunan Restaurant  Indian Restaurant  Italian Restaurant  \
0              0                 0                  0                   0   
1              0                 0                  0                   0   
2              0                 0                  0                   0   
3              0                 0                  0                   0   
4              0                 0                  0                   0   

   Japanese Restaurant  Korean Restaurant  Latin American Restaurant  \
0                    0        

In [144]:
categfreq_data=categ_data.groupby('Community').mean().reset_index()
categfreq_data

Community  African Restaurant  American Restaurant  Arcade  \
0            Armour Square            0.000000             0.166667    0.00   
1                  Ashburn            0.000000             0.500000    0.00   
2           Auburn Gresham            0.000000             0.125000    0.00   
3                   Austin            0.000000             0.000000    0.00   
4              Avalon Park            0.000000             0.000000    0.00   
5                 Avondale            0.000000             0.032258    0.00   
6           Belmont Cragin            0.000000             0.100000    0.00   
7                  Beverly            0.000000             0.000000    0.00   
8               Bridgeport            0.000000             0.100000    0.00   
9            Brighton Park            0.000000             0.000000    0.00   
10                Burnside            0.000000             0.000000    0.00   
11         Calumet Heights            0.000000             0.000000    0.00   
12                 Chatham            0.000000             0.000000    0.00   
13                Clearing            0.000000             0.000000    0.00   
14                 Douglas            0.333333             0.000000    0.00   
15                 Dunning            0.000000             0.125000    0.00   
16      East Garfield Park            0.000000             0.000000    0.00   
17               East Side            0.000000             0.000000    0.00   
18             Edison Park            0.000000             0.000000    0.00   
19               Englewood            0.000000             0.000000    0.00   
20             Forest Glen            0.000000             0.000000    0.00   
21             Fuller Park            0.000000             0.000000    0.00   
22               Gage Park            0.000000             0.000000    0.00   
23          Garfield Ridge            0.000000             0.000000    0.00   
24         Grand Boulevard            0.000000             0.125000    0.00   
25  Greater Grand Crossing            0.000000             0.000000    0.00   
26               Hegewisch            0.000000             0.000000    0.00   
27                 Hermosa            0.000000             0.117647    0.00   
28           Humboldt Park            0.000000             0.000000    0.00   
29               Hyde Park            0.000000             0.071429    0.00   
30             Irving Park            0.000000             0.000000    0.00   
31          Jefferson Park            0.000000             0.000000    0.00   
32                 Kenwood            0.000000             0.000000    0.00   
33            Lincoln Park            0.000000             0.000000    0.00   
34                    Loop            0.000000             0.108696    0.00   
35         Lower West Side            0.000000             0.000000    0.00   
36           McKinley Park            0.000000             0.142857    0.00   
37               Montclare            0.000000             0.000000    0.00   
38             Morgan Park            0.000000             1.000000    0.00   
39         Mount Greenwood            0.000000             0.000000    0.00   
40         Near South Side            0.000000             0.041667    0.00   
41          Near West Side            0.000000             0.111111    0.00   
42            North Center            0.000000             0.105263    0.00   
43          North Lawndale            0.000000             0.000000    0.00   
44              North Park            0.000000             0.076923    0.00   
45            Norwood Park            0.000000             0.000000    0.00   
46                  O'Hare            0.000000             0.000000    0.00   
47                 Oakland            0.000000             0.000000    0.00   
48            Portage Park            0.000000             0.100000    0.00   
49                 Pullman            0.000000             0.000000

In [145]:
categfreq_data.shape

(64, 69)

As we can see above, we have 69 categories of restaurants. That is fine for Clustering model, but it's difficult for human eyes to interpret. To make it easier, let's look at the top 5 categories per community.

In [146]:
# we first define a function that gets top categories
def return_top_categories(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    return row_categories_sorted.index.values[0:num_top_venues]

In [147]:
# now we run a loop that will retrieve top categories for all 64 communities
num_top_venues = 5

indicators = ['st', 'nd', 'rd']
# prepare columns of the dataframe according to our selection of number of top categories that we want to visualize
columns = ['Community'] # Community is our first column, as that is our main reference
for n in np.arange(num_top_venues):
    try:
        columns.append('{}{} Category'.format(n+1, indicators[n]))
    except:
        columns.append('{}th Category'.format(n+1))

# create a new dataframe
top_categ = pd.DataFrame(columns=columns)
top_categ['Community'] = categfreq_data['Community']

for i in np.arange(categfreq_data.shape[0]):
    top_categ.iloc[i, 1:] = return_top_categories(categfreq_data.iloc[i, :], num_top_venues)

top_categ

Community                     1st Category  \
0            Armour Square               Chinese Restaurant   
1                  Ashburn              American Restaurant   
2           Auburn Gresham                             Food   
3                   Austin  Southern / Soul Food Restaurant   
4              Avalon Park                             Food   
5                 Avondale               Mexican Restaurant   
6           Belmont Cragin               Mexican Restaurant   
7                  Beverly               Mexican Restaurant   
8               Bridgeport                             Food   
9            Brighton Park               Mexican Restaurant   
10                Burnside          Comfort Food Restaurant   
11         Calumet Heights                            Diner   
12                 Chatham               Mexican Restaurant   
13                Clearing                             Food   
14                 Douglas               African Restaurant   
15                 Dunning               Mexican Restaurant   
16      East Garfield Park                             Food   
17               East Side                             Food   
18             Edison Park                       Steakhouse   
19               Englewood                             Food   
20             Forest Glen                       Restaurant   
21             Fuller Park             Fast Food Restaurant   
22               Gage Park               Mexican Restaurant   
23          Garfield Ridge               Mexican Restaurant   
24         Grand Boulevard  Southern / Soul Food Restaurant   
25  Greater Grand Crossing                             Food   
26               Hegewisch               Mexican Restaurant   
27                 Hermosa               Mexican Restaurant   
28           Humboldt Park               Mexican Restaurant   
29               Hyde Park                             Food   
30             Irving Park                             Food   
31          Jefferson Park               Chinese Restaurant   
32                 Kenwood               Chinese Restaurant   
33            Lincoln Park               Chinese Restaurant   
34                    Loop                             Food   
35         Lower West Side               Mexican Restaurant   
36           McKinley Park                             Food   
37               Montclare                       Restaurant   
38             Morgan Park              American Restaurant   
39         Mount Greenwood                             Food   
40         Near South Side               Chinese Restaurant   
41          Near West Side                             Food   
42            North Center               Chinese Restaurant   
43          North Lawndale                             Food   
44              North Park                             Food   
45            Norwood Park                      Pizza Place   
46                  O'Hare                              Bar   
47                 Oakland                       Restaurant   
48            Portage Park               Chinese Restaurant   
49                 Pullman               Mexican Restaurant   
50             Rogers Park                             Food   
51                Roseland               Chinese Restaurant   
52           South Chicago               Mexican Restaurant   
53          South Lawndale               Mexican Restaurant   
54             South Shore                             Food   
55                  Uptown            Vietnamese Restaurant   
56         Washington Park              American Restaurant   
57             West Elsdon               Mexican Restaurant   
58          West Englewood                             Food   
59      West Garfield Park                             Food   
60               West Lawn               Mexican Restaurant   
61            West Pullman               Seafood Restaurant   
62               West Town        Latin American

Now let's normalize our 4 Numerical variables - Population, Density, Restaurant Count and a newly calculated ratio Population/Restaurant - by taking each value and dividing by the maximum value of the respective variable. That way, these four variables will fit a range of float values between 0 and 1.

In [148]:
# merge Population, Density and Cont Restaurant data for each community
comb_data=count_rest.merge(community_data, on='Community', how = 'left')
comb_data.head()

Community  Venue ID  Venue Name  Category   Latitude  Longitude  \
0   Armour Square         6           6         6  41.840231 -87.632986   
1         Ashburn         6           6         6  41.747533 -87.711163   
2  Auburn Gresham         8           8         8  41.743387 -87.656042   
3          Austin         6           6         6  41.887876 -87.764851   
4     Avalon Park         5           5         5  41.745035 -87.588658   

   Population  Density (/km2)  
0       13455         5195.00  
1       43792         3479.05  
2       46278         4739.53  
3       95260         5144.07  
4        9985         3084.18

In [149]:
# rename the Venue ID column to be Restaurant count
comb_data=comb_data.rename(columns={'Venue ID': 'Restaurant Count'})

#drop columns that we won't use for clustering
comb_data=comb_data.drop(['Venue Name','Category','Latitude','Longitude'], axis=1)

comb_data.head()

Community  Restaurant Count  Population  Density (/km2)
0   Armour Square                 6       13455         5195.00
1         Ashburn                 6       43792         3479.05
2  Auburn Gresham                 8       46278         4739.53
3          Austin                 6       95260         5144.07
4     Avalon Park                 5        9985         3084.18

In [150]:
# Now let's add a column that calculates average number of people per restaurant
comb_data['People/Restaurant']=comb_data['Population']/comb_data['Restaurant Count']
comb_data.head()

Community  Restaurant Count  Population  Density (/km2)  \
0   Armour Square                 6       13455         5195.00   
1         Ashburn                 6       43792         3479.05   
2  Auburn Gresham                 8       46278         4739.53   
3          Austin                 6       95260         5144.07   
4     Avalon Park                 5        9985         3084.18   

   People/Restaurant  
0        2242.500000  
1        7298.666667  
2        5784.750000  
3       15876.666667  
4        1997.000000

Now let's merge this dataset with the category frequency counts for each community

In [151]:
comb_data=comb_data.merge(categfreq_data, on='Community', how = 'right')
comb_data.head()

Community  Restaurant Count  Population  Density (/km2)  \
0   Armour Square                 6       13455         5195.00   
1         Ashburn                 6       43792         3479.05   
2  Auburn Gresham                 8       46278         4739.53   
3          Austin                 6       95260         5144.07   
4     Avalon Park                 5        9985         3084.18   

   People/Restaurant  African Restaurant  American Restaurant  Arcade  \
0        2242.500000                 0.0             0.166667     0.0   
1        7298.666667                 0.0             0.500000     0.0   
2        5784.750000                 0.0             0.125000     0.0   
3       15876.666667                 0.0             0.000000     0.0   
4        1997.000000                 0.0             0.000000     0.0   

   Argentinian Restaurant  Asian Restaurant  BBQ Joint  Bagel Shop  Bakery  \
0                     0.0               0.0        0.0         0.0     0.0   
1                     0.0               0.0        0.0         0.0     0.0   
2                     0.0               0.0        0.0         0.0     0.0   
3                     0.0               0.0        0.0         0.0     0.0   
4                     0.0               0.0        0.0         0.0     0.0   

   Bar  Breakfast Spot  Burger Joint  Café  Cajun / Creole Restaurant  \
0  0.0        0.000000           0.0   0.0                        0.0   
1  0.0        0.000000           0.0   0.0                        0.0   
2  0.0        0.000000           0.0   0.0                        0.0   
3  0.0        0.166667           0.0   0.0                        0.0   
4  0.0        0.000000           0.0   0.0                        0.0   

   Cantonese Restaurant  Caribbean Restaurant  Chinese Restaurant  \
0                   0.0                   0.0               0.500   
1                   0.0                   0.0               0.000   
2                   0.0                   0.0               0.125   
3                   0.0                   0.0               0.000   
4                   0.0                   0.0               0.000   

   Coffee Shop  Comfort Food Restaurant  Cuban Restaurant  Deli / Bodega  \
0          0.0                      0.0               0.0            0.0   
1          0.0                      0.0               0.0            0.0   
2          0.0                      0.0               0.0            0.0   
3          0.0                      0.0               0.0            0.0   
4          0.0                      0.2               0.0            0.0   

   Dim Sum Restaurant  Diner  Eastern European Restaurant  \
0                 0.0    0.0                          0.0   
1                 0.0    0.0                          0.0   
2                 0.0    0.0                          0.0   
3                 0.0    0.0                          0.0   
4                 0.0    0.2                          0.0   

   Ethiopian Restaurant  Fast Food Restaurant      Food  Food Court  \
0                   0.0              0.166667  0.166667         0.0   
1                   0.0              0.000000  0.000000         0.0   
2                   0.0              0.000000  0.375000         0.0   
3                   0.0              0.000000  0.166667         0.0   
4                   0.0              0.200000  0.400000         0.0   

   French Restaurant  Gastropub  Gay Bar  German Restaurant  Greek Restaurant  \
0                0.0        0.0      0.0                0.0          0.000000   
1                0.0        0.0      0.0                0.0          0.000000   
2                0.0        0.0      0.0                0.0          0.000000   
3                0.0        0.0      0.0                0.0          0.166667   
4                0.0        0.0      0.0                0.0          0.000000   

   Hot Dog Joint  Hunan Restaurant  Indian Restaurant  Italian Restaurant  \
0       0.000000            

In [152]:
comb_data.shape

(64, 73)

Now let's just normalize columns 1-4 by dividing each value by the maximum value in each column

In [159]:
# create new dataframe cluster_data, initiating it as copy of comb_data
cluster_columns = comb_data.columns.tolist()
cluster_data = pd.DataFrame (columns = cluster_columns)
cluster_data = comb_data.copy()

cluster_data['Restaurant Count']=cluster_data['Restaurant Count']/cluster_data['Restaurant Count'].max()
cluster_data['Population']=cluster_data['Population']/cluster_data['Population'].max()
cluster_data['Density (/km2)']=cluster_data['Density (/km2)']/cluster_data['Density (/km2)'].max()
cluster_data['People/Restaurant']=cluster_data['People/Restaurant']/cluster_data['People/Restaurant'].max()
cluster_data

Community  Restaurant Count  Population  Density (/km2)  \
0            Armour Square          0.125000    0.141245        0.449624   
1                  Ashburn          0.125000    0.459710        0.301109   
2           Auburn Gresham          0.166667    0.485807        0.410203   
3                   Austin          0.125000    1.000000        0.445216   
4              Avalon Park          0.104167    0.104818        0.266934   
5                 Avondale          0.645833    0.392274        0.630667   
6           Belmont Cragin          0.416667    0.838862        0.682952   
7                  Beverly          0.020833    0.218581        0.218807   
8               Bridgeport          0.208333    0.353107        0.537820   
9            Brighton Park          0.145833    0.470428        0.550556   
10                Burnside          0.020833    0.023662        0.123478   
11         Calumet Heights          0.062500    0.138442        0.251830   
12                 Chatham          0.062500    0.326685        0.352520   
13                Clearing          0.083333    0.271793        0.339292   
14                 Douglas          0.062500    0.218150        0.420870   
15                 Dunning          0.166667    0.458629        0.392459   
16      East Garfield Park          0.083333    0.209910        0.346220   
17               East Side          0.062500    0.249181        0.266180   
18             Edison Park          0.020833    0.121824        0.141534   
19               Englewood          0.062500    0.263227        0.272941   
20             Forest Glen          0.083333    0.199654        0.198612   
21             Fuller Park          0.083333    0.025604        0.114794   
22               Gage Park          0.208333    0.429068        0.620840   
23          Garfield Ridge          0.125000    0.382070        0.287527   
24         Grand Boulevard          0.166667    0.234233        0.428523   
25  Greater Grand Crossing          0.083333    0.333466        0.299020   
26               Hegewisch          0.020833    0.098866        0.060061   
27                 Hermosa          0.354167    0.253454        0.689588   
28           Humboldt Park          0.125000    0.592347        0.523782   
29               Hyde Park          0.291667    0.281619        0.556817   
30             Irving Park          0.395833    0.573231        0.568461   
31          Jefferson Park          0.145833    0.281419        0.384481   
32                 Kenwood          0.062500    0.180443        0.552310   
33            Lincoln Park          0.312500    0.710792        0.716031   
34                    Loop          0.958333    0.376653        0.726665   
35         Lower West Side          0.437500    0.345245        0.375090   
36           McKinley Park          0.145833    0.165515        0.373677   
37               Montclare          0.104167    0.145182        0.466823   
38             Morgan Park          0.020833    0.235083        0.226769   
39         Mount Greenwood          0.083333    0.202362        0.237703   
40         Near South Side          1.000000    0.247953        0.443430   
41          Near West Side          0.187500    0.660004        0.369242   
42            North Center          0.395833    0.375698        0.583393   
43          North Lawndale          0.041667    0.377357        0.374217   
44              North Park          0.270833    0.197796        0.249857   
45            Norwood Park          0.145833    0.389345        0.283615   
46                  O'Hare          0.125000    0.129929        0.031005   
47                 Oakland          0.083333    0.069756        0.382853   
48            Portage Park          0.208333    0.675068        0.544035   
49                 Pullman          0.083333    0.069421        0.104238   
50             Rogers Park          0.416667    0.578018        1.000000   
51                Roseland          0.062500    0.445444        0

### 2b. Running the Clustering model

#### The number K of clusters is a key parameter when preparing a Clustering model. After a few iterations testing K numbers from 3 to 8, I found that 5 clusters gave the best results. 3 or 4 clusters didn't yield a clear profile of clusters, while 6-8 clusters resulted in two or more clusters with only 1 or 2 communities each. Therefore, we will proceed with K=5.

In [160]:
# set number of clusters
kclusters = 5

X = cluster_data.drop('Community', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(X)

# check cluster labels generated for each row in the dataframe
kmeans.labels_

array([4, 0, 3, 2, 3, 2, 2, 1, 3, 1, 4, 1, 1, 3, 3, 1, 3, 3, 4, 3, 4, 4,
       1, 1, 4, 3, 1, 2, 1, 2, 2, 4, 4, 2, 2, 1, 3, 4, 0, 3, 2, 3, 2, 3,
       3, 3, 4, 4, 2, 4, 2, 4, 1, 1, 2, 2, 4, 1, 3, 3, 1, 0, 2, 4],
      dtype=int32)

In [161]:
# Confirms that all 64 communities got a cluster label 
kmeans.labels_.shape[0]

64

Now for our final analysis, let's create a data frame that merges initial community information, top 5 restaurant categories, and their cluster labels

In [162]:
# initiate a dataframe using Community name as key field
final_data=pd.DataFrame(columns={'Community'})
final_data['Community']=cluster_data['Community']

# add cluster labels
final_data.insert(1, 'Cluster Labels', kmeans.labels_)

# add columns Restaurant count, Population, Density and People/Restaurant columns from comb_data
subset=comb_data.iloc[:,0:5]
final_data=final_data.merge(subset, on='Community', how = 'right')

# add top 5 restaurant categories for each community
final_data=final_data.merge(top_categ, on='Community', how = 'right')

# add Longitude and Latitude at the end of the table
coordinates_subset = community_data[['Community','Latitude','Longitude']]
final_data=final_data.merge(coordinates_subset, on='Community', how = 'left')

print(final_data.shape)
final_data

(64, 13)


Community  Cluster Labels  Restaurant Count  Population  \
0            Armour Square               4                 6       13455   
1                  Ashburn               0                 6       43792   
2           Auburn Gresham               3                 8       46278   
3                   Austin               2                 6       95260   
4              Avalon Park               3                 5        9985   
5                 Avondale               2                31       37368   
6           Belmont Cragin               2                20       79910   
7                  Beverly               1                 1       20822   
8               Bridgeport               3                10       33637   
9            Brighton Park               1                 7       44813   
10                Burnside               4                 1        2254   
11         Calumet Heights               1                 3       13188   
12                 Chatham               1                 3       31120   
13                Clearing               3                 4       25891   
14                 Douglas               3                 3       20781   
15                 Dunning               1                 8       43689   
16      East Garfield Park               3                 4       19996   
17               East Side               3                 3       23737   
18             Edison Park               4                 1       11605   
19               Englewood               3                 3       25075   
20             Forest Glen               4                 4       19019   
21             Fuller Park               4                 4        2439   
22               Gage Park               1                10       40873   
23          Garfield Ridge               1                 6       36396   
24         Grand Boulevard               4                 8       22313   
25  Greater Grand Crossing               3                 4       31766   
26               Hegewisch               1                 1        9418   
27                 Hermosa               2                17       24144   
28           Humboldt Park               1                 6       56427   
29               Hyde Park               2                14       26827   
30             Irving Park               2                19       54606   
31          Jefferson Park               4                 7       26808   
32                 Kenwood               4                 3       17189   
33            Lincoln Park               2                15       67710   
34                    Loop               2                46       35880   
35         Lower West Side               1                21       32888   
36           McKinley Park               3                 7       15767   
37               Montclare               4                 5       13830   
38             Morgan Park               0                 1       22394   
39         Mount Greenwood               3                 4       19277   
40         Near South Side               2                48       23620   
41          Near West Side               3                 9       62872   
42            North Center               2                19       35789   
43          North Lawndale               3                 2       35947   
44              North Park               3                13       18842   
45            Norwood Park               3                 7       37089   
46                  O'Hare               4                 6       12377   
47                 Oakland               4                 4        6645   
48            Portage Park               2                10       64307   
49                 Pullman               4                 4        6613   
50             Rogers Park               2                20       55062   
51                Roseland               4                 3     

### 2c. Visualizing and analyzing community clusters

Let's first visualize our clusters on a Chicago map

In [163]:
# create map
map_clusters = folium.Map(location=[chicago_latitude, chicago_longitude], zoom_start=10)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, com, cluster in zip(final_data['Latitude'], final_data['Longitude'], final_data['Community'], final_data['Cluster Labels']):
    label = folium.Popup(' Cluster ' + str(cluster) + ', ' + str(com) , parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

Let's now analyze the characteristics of each cluster. We will start by analyzing statistics of numerical data accross all of them to use as reference when looking at each of them separately.

In [164]:
# start by taking a quick look at the statistics of all numerical data: Restaurant Count, Population, Density and People/Restaurant
subset=final_data.iloc[:,2:6]
subset.describe()

Restaurant Count    Population  Density (/km2)  People/Restaurant
count         64.000000     64.000000       64.000000          64.000000
mean           9.796875  32471.546875     4676.187500        5781.675779
std           10.486847  20489.634950     2191.099279        5379.199691
min            1.000000   2254.000000      358.230000         492.083333
25%            4.000000  18974.750000     3136.237500        2249.946429
50%            6.000000  27284.500000     4336.920000        4455.000000
75%           11.500000  42747.000000     6309.310000        6941.694444
max           48.000000  95260.000000    11554.110000       27742.000000

Now let's look at features of each cluster separately.

In [165]:
cluster_number=0
count=final_data[final_data['Cluster Labels']==cluster_number].nunique()[3]
print("There are {} communities in Cluster number {}.".format(count, cluster_number))
final_data[final_data['Cluster Labels']==cluster_number]

There are 3 communities in Cluster number 0.


Community  Cluster Labels  Restaurant Count  Population  \
1        Ashburn               0                 6       43792   
38   Morgan Park               0                 1       22394   
61  West Pullman               0                 1       27742   

    Density (/km2)  People/Restaurant         1st Category       2nd Category  \
1          3479.05        7298.666667  American Restaurant         Sports Bar   
38         2620.11       22394.000000  American Restaurant  Yemeni Restaurant   
61         3008.78       27742.000000   Seafood Restaurant  Yemeni Restaurant   

            3rd Category        4th Category        5th Category   Latitude  \
1          Hot Dog Joint  Italian Restaurant   Yemeni Restaurant  41.747533   
38                  Food       Deli / Bodega  Dim Sum Restaurant  41.690312   
61  Ethiopian Restaurant    Cuban Restaurant       Deli / Bodega  41.671775   

    Longitude  
1  -87.711163  
38 -87.666716  
61 -87.638358

The first cluster (Cluster 0) has 3 communities with a high ratio of People/Restaurant compared to the entire set of 64 clustered communities (above 3rd quartile = 6,941 people/restaurant). Those communities are also located far from Chicago downtown and, based on the list of top 5 categories in those communities, there is no indication that a French restaurant would be particularly successful there.

In [166]:
cluster_number=1
count=final_data[final_data['Cluster Labels']==cluster_number].nunique()[3]
print("There are {} communities in Cluster number {}.".format(count, cluster_number))
final_data[final_data['Cluster Labels']==cluster_number]

There are 14 communities in Cluster number 1.


Community  Cluster Labels  Restaurant Count  Population  \
7           Beverly               1                 1       20822   
9     Brighton Park               1                 7       44813   
11  Calumet Heights               1                 3       13188   
12          Chatham               1                 3       31120   
15          Dunning               1                 8       43689   
22        Gage Park               1                10       40873   
23   Garfield Ridge               1                 6       36396   
26        Hegewisch               1                 1        9418   
28    Humboldt Park               1                 6       56427   
35  Lower West Side               1                21       32888   
52    South Chicago               1                 7       28263   
53   South Lawndale               1                16       74851   
57      West Elsdon               1                 9       19237   
60        West Lawn               1                14       33108   

    Density (/km2)  People/Restaurant        1st Category  \
7          2528.12       20822.000000  Mexican Restaurant   
9          6361.18        6401.857143  Mexican Restaurant   
11         2909.67        4396.000000               Diner   
12         4073.05       10373.333333  Mexican Restaurant   
15         4534.52        5461.125000  Mexican Restaurant   
22         7173.25        4087.300000  Mexican Restaurant   
23         3322.12        6066.000000  Mexican Restaurant   
26          693.95        9418.000000  Mexican Restaurant   
28         6051.83        9404.500000  Mexican Restaurant   
35         4333.83        1566.095238  Mexican Restaurant   
52         3267.19        4037.571429  Mexican Restaurant   
53         6296.33        4678.187500  Mexican Restaurant   
57         6348.25        2137.444444  Mexican Restaurant   
60         4333.24        2364.857143  Mexican Restaurant   

                   2nd Category          3rd Category          4th Category  \
7             Yemeni Restaurant  Fast Food Restaurant      Cuban Restaurant   
9            Seafood Restaurant    Chinese Restaurant        Breakfast Spot   
11                 Burger Joint    Mexican Restaurant     Yemeni Restaurant   
12         Caribbean Restaurant     Yemeni Restaurant  Fast Food Restaurant   
15  Eastern European Restaurant                   Bar   American Restaurant   
22             Asian Restaurant           Coffee Shop               Gay Bar   
23                   Restaurant                  Food  Ethiopian Restaurant   
26            Yemeni Restaurant  Fast Food Restaurant      Cuban Restaurant   
28         Caribbean Restaurant            Taco Place        Breakfast Spot   
35                         Food            Restaurant            Taco Place   
52           Italian Restaurant  Caribbean Restaurant                  Food   
53           Chinese Restaurant                 Diner            Restaurant   
57                         Food   American Restaurant    Chinese Restaurant   
60                  Pizza Place    Seafood Restaurant                  Food   

            5th Category   Latitude  Longitude  
7          Deli / Bodega  41.718153 -87.671767  
9          Deli / Bodega  41.818922 -87.698942  
11  Fast Food Restaurant  41.730035 -87.579213  
12         Deli / Bodega  41.741145 -87.612548  
15    Seafood Restaurant  41.952809 -87.796449  
22             Gastropub  41.795033 -87.696164  
23      Cuban Restaurant  41.803617 -87.745489  
26         Deli / Bodega  41.653646 -87.546988  
28  Fast Food Restaurant  41.900828 -87.723959  
35            Food Court  41.854200 -87.665609  
52    African Restaurant  41.734587 -87.551322  
53     Yemeni Restaurant  41.843644 -87.712554  
57         Deli / Bodega  41.793922 -87.724498  
60    Chinese Restaurant  41.772811 -87.722275

Cluster 1 seems to feature high interest for Mexican food, as those are the most frequent categories of restaurants open in the region. It also seems that Caribbean, American and Fast Food are popular cuisines in those locations. I also noticed that most communities (10 out of 14) have Population values above the average of 27,284 people per community, seen across all 64 analyzed communities.

In [167]:
cluster_number=2
count=final_data[final_data['Cluster Labels']==cluster_number].nunique()[3]
print("There are {} communities in Cluster number {}.".format(count, cluster_number))
final_data[final_data['Cluster Labels']==cluster_number]

There are 15 communities in Cluster number 2.


Community  Cluster Labels  Restaurant Count  Population  \
3            Austin               2                 6       95260   
5          Avondale               2                31       37368   
6    Belmont Cragin               2                20       79910   
27          Hermosa               2                17       24144   
29        Hyde Park               2                14       26827   
30      Irving Park               2                19       54606   
33     Lincoln Park               2                15       67710   
34             Loop               2                46       35880   
40  Near South Side               2                48       23620   
42     North Center               2                19       35789   
48     Portage Park               2                10       64307   
50      Rogers Park               2                20       55062   
54      South Shore               2                11       50418   
55           Uptown               2                47       57973   
62        West Town               2                25       84502   

    Density (/km2)  People/Restaurant                     1st Category  \
3          5144.07       15876.666667  Southern / Soul Food Restaurant   
5          7286.80        1205.419355               Mexican Restaurant   
6          7890.90        3995.500000               Mexican Restaurant   
27         7967.57        1420.235294               Mexican Restaurant   
29         6433.52        1916.214286                             Food   
30         6568.06        2874.000000                             Food   
33         8273.10        4514.000000               Chinese Restaurant   
34         8395.97         780.000000                             Food   
40         5123.44         492.083333               Chinese Restaurant   
42         6740.59        1883.631579               Chinese Restaurant   
48         6285.84        6430.700000               Chinese Restaurant   
50        11554.11        2753.100000                             Food   
54         6643.86        4583.454545                             Food   
55         9648.06        1233.468085            Vietnamese Restaurant   
62         7123.67        3380.080000        Latin American Restaurant   

           2nd Category               3rd Category             4th Category  \
3      Greek Restaurant             Breakfast Spot                     Food   
5    Chinese Restaurant                       Food    Vietnamese Restaurant   
6            Restaurant        American Restaurant                     Food   
27           Restaurant  Latin American Restaurant      American Restaurant   
29  Japanese Restaurant         Mexican Restaurant      American Restaurant   
30          Pizza Place         Mexican Restaurant       Chinese Restaurant   
33           Bagel Shop               Noodle House  Scandinavian Restaurant   
34   Chinese Restaurant        American Restaurant       Italian Restaurant   
40                 Food           Asian Restaurant      American Restaurant   
42      Thai Restaurant  Latin American Restaurant      American Restaurant   
48           Restaurant         Italian Restaurant       Mexican Restaurant   
50   Mexican Restaurant         Chinese Restaurant      American Restaurant   
54   African Restaurant             Breakfast Spot                  Gay Bar   
55                 Food         Chinese Restaurant       African Restaurant   
62   Mexican Restaurant       Caribbean Restaurant     Ukrainian Restaurant   

                   5th Category   Latitude  Longitude  
3                    Taco Place  41.887876 -87.764851  
5                         Diner  41.938921 -87.711168  
6   Eastern European Restaurant  41.931698 -87.768670  
27                  Pizza Place  41.928643 -87.734502  
29                    Gastropub  41.794446 -87.593924  
30     Mediterranean Restaurant  41.953365 -87.736447  
33           Seafood Restaurant  41.939945 -87.636120  
34     

Cluster number 2 gathers 15 communities with high Population and Population Density, compared to the total averages seen across all 64 communities. It also comprises a set of communities with very diverse cuisines, which indicates that restaurant guests in the area probably appreciate having variety of food and dining experiences. This is also the cluster where European cuisine is more frequently seen in top 5 categories: Greek, Eastern European, Scandinavian, Italian, and Ukrainian cuisines are featured in the top 5 categories. This could be an indication that French cuisine – also European – could be successful in the location.

In [168]:
cluster_number=3
count=final_data[final_data['Cluster Labels']==cluster_number].nunique()[3]
print("There are {} communities in Cluster number {}.".format(count, cluster_number))
final_data[final_data['Cluster Labels']==cluster_number]

There are 17 communities in Cluster number 3.


Community  Cluster Labels  Restaurant Count  Population  \
2           Auburn Gresham               3                 8       46278   
4              Avalon Park               3                 5        9985   
8               Bridgeport               3                10       33637   
13                Clearing               3                 4       25891   
14                 Douglas               3                 3       20781   
16      East Garfield Park               3                 4       19996   
17               East Side               3                 3       23737   
19               Englewood               3                 3       25075   
25  Greater Grand Crossing               3                 4       31766   
36           McKinley Park               3                 7       15767   
39         Mount Greenwood               3                 4       19277   
41          Near West Side               3                 9       62872   
43          North Lawndale               3                 2       35947   
44              North Park               3                13       18842   
45            Norwood Park               3                 7       37089   
58          West Englewood               3                 4       29929   
59      West Garfield Park               3                 5       17163   

    Density (/km2)  People/Restaurant        1st Category  \
2          4739.53        5784.750000                Food   
4          3084.18        1997.000000                Food   
8          6214.03        3363.700000                Food   
13         3920.22        6472.750000                Food   
14         4862.78        6927.000000  African Restaurant   
16         4000.26        4999.000000                Food   
17         3075.47        7912.333333                Food   
19         3153.59        8358.333333                Food   
25         3454.91        7941.500000                Food   
36         4317.50        2252.428571                Food   
39         2746.45        4819.250000                Food   
41         4266.26        6985.777778                Food   
43         4323.74       17973.500000                Food   
44         2886.88        1449.384615                Food   
45         3276.92        5298.428571         Pizza Place   
58         3668.46        7482.250000                Food   
59         5177.09        3432.600000                Food   

                       2nd Category               3rd Category  \
2   Southern / Soul Food Restaurant         Chinese Restaurant   
4              Fast Food Restaurant                      Diner   
8               American Restaurant                 Restaurant   
13                      Pizza Place           Asian Restaurant   
14                             Café                       Food   
16                            Diner       Fast Food Restaurant   
17                              Bar         Italian Restaurant   
19                       Restaurant       Ethiopian Restaurant   
25  Southern / Soul Food Restaurant          Yemeni Restaurant   
36                            Diner        American Restaurant   
39                   Breakfast Spot         Italian Restaurant   
41               Mexican Restaurant       Fast Food Restaurant   
43             Fast Food Restaurant           Cuban Restaurant   
44                Korean Restaurant          Yemeni Restaurant   
45                            Diner  Latin American Restaurant   
58              American Restaurant       Fast Food Restaurant   
59                       Taco Place                Coffee Shop   

                   4th Category          5th Category   Latitude  Longitude  
2            Mexican Restaurant   American Restaurant  41.743387 -87.656042  
4       Comfort Food Restaurant               Gay Bar  41.745035 -87.588658  
8              Asian Restaurant   Japanese Restaurant  41.837938 -87.651028  
13    Latin American Restaurant  Fast Food Resta

Cluster number 3 features varying values of Population, Density and People/Restaurant ratio across communities, hence I we cannot see a pattern regarding those numerical features. Cuisines in this area seem concentrated around General and American Food, Fast Food and Delis, as well as Cuban and Asian Food. Those cuisines are generally very different from French cuisine, so this is probably not a strong location for a French restaurant. 

In [169]:
cluster_number=4
count=final_data[final_data['Cluster Labels']==cluster_number].nunique()[3]
print("There are {} communities in Cluster number {}.".format(count, cluster_number))
final_data[final_data['Cluster Labels']==cluster_number]

There are 15 communities in Cluster number 4.


Community  Cluster Labels  Restaurant Count  Population  \
0     Armour Square               4                 6       13455   
10         Burnside               4                 1        2254   
18      Edison Park               4                 1       11605   
20      Forest Glen               4                 4       19019   
21      Fuller Park               4                 4        2439   
24  Grand Boulevard               4                 8       22313   
31   Jefferson Park               4                 7       26808   
32          Kenwood               4                 3       17189   
37        Montclare               4                 5       13830   
46           O'Hare               4                 6       12377   
47          Oakland               4                 4        6645   
49          Pullman               4                 4        6613   
51         Roseland               4                 3       42433   
56  Washington Park               4                 3       11502   
63         Woodlawn               4                 5       23268   

    Density (/km2)  People/Restaurant                     1st Category  \
0          5195.00        2242.500000               Chinese Restaurant   
10         1426.68        2254.000000          Comfort Food Restaurant   
18         1635.30       11605.000000                       Steakhouse   
20         2294.78        4754.750000                       Restaurant   
21         1326.34         609.750000             Fast Food Restaurant   
24         4951.20        2789.125000  Southern / Soul Food Restaurant   
31         4442.33        3829.714286               Chinese Restaurant   
32         6381.45        5729.666667               Chinese Restaurant   
37         5393.73        2766.000000                       Restaurant   
46          358.23        2062.833333                              Bar   
47         4423.53        1661.250000                       Restaurant   
49         1204.38        1653.250000               Mexican Restaurant   
51         3399.06       14144.333333               Chinese Restaurant   
56         2921.68        3834.000000              American Restaurant   
63         4340.01        4653.600000             Fast Food Restaurant   

                       2nd Category               3rd Category  \
0                              Food       Fast Food Restaurant   
10             Fast Food Restaurant           Cuban Restaurant   
18                Yemeni Restaurant       Ethiopian Restaurant   
20                Indian Restaurant         Chinese Restaurant   
21               Italian Restaurant                      Diner   
24               Chinese Restaurant                  BBQ Joint   
31                       Restaurant  Middle Eastern Restaurant   
32             Caribbean Restaurant                  BBQ Joint   
37               Mexican Restaurant                Pizza Place   
46                       Sports Bar           Greek Restaurant   
47             Caribbean Restaurant                  BBQ Joint   
49             Fast Food Restaurant                      Diner   
51  Southern / Soul Food Restaurant       Fast Food Restaurant   
56               Chinese Restaurant                 Restaurant   
63              American Restaurant       Caribbean Restaurant   

                       4th Category                 5th Category   Latitude  \
0               American Restaurant                Deli / Bodega  41.840231   
10                    Deli / Bodega           Dim Sum Restaurant  41.730035   
18                 Cuban Restaurant                Deli / Bodega  42.006113   
20                Yemeni Restaurant         Ethiopian Restaurant  41.991752   
21  Southern / Soul Food Restaurant         Ethiopian Restaurant  41.818089   
24                       Restaurant         Caribbean Restaurant  41.813923   
31                             Food  Eastern European Restaurant  41.969738   
32                Yemeni Restau

The last Cluster, number 4, comprises communities where there seems to be a concentration of American, Southern and Fast Food, as well as Chinese, Middle Eastern and African Food. The Population in most of those communities are below average of the entire community dataset. Similarly, the average number of people per restaurant seems generally below average of the entire dataset. This may be an indication that the area is already filled with many restaurants and opening a new dining venue there could be more challenging.

### 3. Conclusion

#### Based on the analysis above, it seems that Cluster number 2 is the most promising set of communities for a new French Restaurant: it gathers areas where a broad variety of cuisines make the top 5 categories, and the only cluster where European restaurants have more prominent presence. Moreover, communities in this cluster have large Population and Population Density, which increases chances of higher daily occupancy rates, a crucial metric for financially sustaining restaurants anywhere.